File: ai-service/schemas/feedback.py
from pydantic import BaseModel, Field, ConfigDict
from typing import Optional, Literal
from datetime import datetime

class FeedbackBase(BaseModel):
    type: Optional[Literal['Bug Report', 'Feature Request', 'General', 'Caption Quality']] = 'General'
    subject: Optional[str] = Field(None, max_length=120)
    message: str
    rating: Optional[int] = Field(None, ge=1, le=5)
    screenshot_url: Optional[str] = None

class FeedbackCreate(FeedbackBase):
    pass

class FeedbackUpdate(BaseModel):
    type: Optional[Literal['Bug Report', 'Feature Request', 'General', 'Caption Quality']] = None
    subject: Optional[str] = None
    message: Optional[str] = None
    rating: Optional[int] = Field(None, ge=1, le=5)
    screenshot_url: Optional[str] = None

class FeedbackAdminUpdate(BaseModel):
    status: Optional[Literal['New', 'In Review', 'Resolved', "Won't Fix"]] = None

class FeedbackOut(FeedbackBase):
    id: int
    user_id: int
    status: str
    created_at: datetime
    updated_at: datetime
    
    model_config = ConfigDict(from_attributes=True)

class FeedbackStats(BaseModel):
    total_count: int
    average_rating: float
    status_breakdown: dict
    type_distribution: dict

File: frontend/src/types/feedback.ts
export type FeedbackStatus = 'New' | 'In Review' | 'Resolved' | "Won't Fix";
export type FeedbackType = 'Bug Report' | 'Feature Request' | 'General' | 'Caption Quality';

export interface Feedback {
  id: number;
  user_id: number;
  type: FeedbackType;
  subject: string | null;
  message: string;
  rating: number | null;
  status: FeedbackStatus;
  screenshot_url: string | null;
  created_at: string;
  updated_at: string;
}

export interface FeedbackCreateInput {
  type?: FeedbackType;
  subject?: string;
  message: string;
  rating?: number;
  screenshot_url?: string;
}

export interface FeedbackAdminUpdateInput {
  status?: FeedbackStatus;
}

export interface FeedbackUpdateInput {
  type?: FeedbackType;
  subject?: string;
  message?: string;
  rating?: number;
  screenshot_url?: string;
}

export interface FeedbackStatsData {
  total_count: number;
  average_rating: number;
  status_breakdown: Record<string, number>;
  type_distribution: Record<string, number>;
}